core/src/epicli/cli/engine/terraform/TerraformRunner.py

```diff
@@ -52,22 +52,29 @@ def azure_login(self):
                 # Create the service principal, for now we use the default subscription
                 self.logger.info('Creating service principal')
                 cluster_name = self.cluster_model.specification.name.lower()
-                cluster_prefix = self.cluster_model.specification.prefix.lower()               
+                cluster_prefix = self.cluster_model.specification.prefix.lower()
                 resource_group_name = resource_name(cluster_prefix, cluster_name, 'rg')
                 sp = apiproxy.create_sp(resource_group_name, subscription['id'])
                 sp['subscriptionId'] = subscription['id']
                 save_sp(sp, self.cluster_model.specification.name)
             else:
-                self.logger.info('Using service principal from file')
+                self.logger.info(f'Using service principal from file {sp_file}')
                 sp = load_yaml_file(sp_file)
 
             # Login as SP and get the default subscription.
             subscription = apiproxy.login_sp(sp)
 
             if 'subscriptionId' in sp:
-                # Set active subscription if sp contains it. 
+                # Set active subscription if sp contains it.
+                self.logger.debug(f"subscriptionId from sp file: {sp['subscriptionId']}")
                 apiproxy.set_active_subscribtion(sp['subscriptionId'])
-                self.new_env['ARM_SUBSCRIPTION_ID'] = sp['subscriptionId']
+                arm_subscription_id = ""
+                if "/" in sp['subscriptionId']:
+                    self.logger.debug(f"WARN Slash detected in the subscription name {sp['subscriptionId']}. Will parse the ID and use it instead")
+                    arm_subscription_id = [x['id'] for x in subscription if x['name'] == sp['subscriptionId']][0]
```

Comment:
What's about usage of [generator expressions](https://www.python.org/dev/peps/pep-0289/) here? Something like this:

```
arm_subscription_id = next((x['id'] for x in subscription if x['name'] == sp['subscriptionId']), None)
```

Please change the name of "x" to something valuable.

Reply:
Interesting new feature, I start looking into it and see if it work. brb

Reply:
Seems to work
```
In [1]: l0 = [1,2,3,4,5]                                                                                                               

In [2]: y = next(x for x in l0 if x == 4)                                                                                              

In [3]: y                                                                                                                              
Out[3]: 4

In [4]: y = next((x for x in l0 if x == 4), None)                                                                                      

In [5]: y                                                                                                                              
Out[5]: 4
```